manim/utils/color/core.py

```diff
@@ -689,34 +819,165 @@ def __eq__(self, other: object) -> bool:
+        )
 
     def __or__(self, other: ManimColor) -> ManimColor:
-        return ManimColor(self.to_integer() | other.to_integer())
+        return self._construct_from_space(
+            ManimColor._internal_from_integer(
+                self.to_integer() | other.to_integer(), 1.0
+            )
+        )
 
     def __xor__(self, other: ManimColor) -> ManimColor:
-        return ManimColor(self.to_integer() ^ other.to_integer())
+        return self._construct_from_space(
+            ManimColor._internal_from_integer(
+                self.to_integer() ^ other.to_integer(), 1.0
+            )
+        )
+
+
+RGBA = ManimColor
+"""RGBA Color Space"""
+
+
+class HSV(ManimColor):
+    """HSV Color Space"""
+
+    def __init__(
+        self,
+        hsv: HSV_Array_Float | HSV_Tuple_Float | HSVA_Array_Float | HSVA_Tuple_Float,
+        alpha: float = 1.0,
+    ) -> None:
+        super().__init__(None)
+        if len(hsv) == 3:
+            self.__hsv: HSVA_Array_Float = np.asarray((*hsv, alpha))
+        elif len(hsv) == 4:
+            self.__hsv: HSVA_Array_Float = np.asarray(hsv)
+        else:
+            raise ValueError("HSV Color must be an array of 3 values")
+
+    @classmethod
+    @override
+    def _from_internal(cls, value: ManimColorInternal) -> Self:
+        hsv = colorsys.rgb_to_hsv(*value[:3])
+        hsva = [*hsv, value[-1]]
+        return cls(np.array(hsva))
+
+    @property
+    def hue(self) -> float:
+        return self.__hsv[0]
+
+    @property
+    def saturation(self) -> float:
+        return self.__hsv[1]
+
+    @property
+    def value(self) -> float:
+        return self.__hsv[2]
+
+    @hue.setter
+    def hue(self, value: float) -> None:
+        self.__hsv[0] = value
+
+    @saturation.setter
+    def saturation(self, value: float) -> None:
+        self.__hsv[1] = value
+
+    @value.setter
+    def value(self, value: float) -> None:
+        self.__hsv[2] = value
+
+    @property
+    def h(self) -> float:
+        return self.__hsv[0]
+
+    @property
+    def s(self) -> float:
+        return self.__hsv[1]
+
+    @property
+    def v(self) -> float:
+        return self.__hsv[2]
+
+    @h.setter
+    def h(self, value: float) -> None:
+        self.__hsv[0] = value
+
+    @s.setter
+    def s(self, value: float) -> None:
+        self.__hsv[1] = value
+
+    @v.setter
+    def v(self, value: float) -> None:
+        self.__hsv[2] = value
+
+    @property
+    def _internal_space(self) -> npt.NDArray:
+        return self.__hsv
+
+    @property
+    def _internal_value(self) -> ManimColorInternal:
+        """Returns the internal value of the current Manim color [r,g,b,a] float array
+
+        Returns
+        -------
+        ManimColorInternal
+            internal color representation
+        """
+        return np.array(
+            [
+                *colorsys.hsv_to_rgb(self.__hsv[0], self.__hsv[1], self.__hsv[2]),
+                self.__alpha,
+            ],
+            dtype=float,
```

Comment:
```suggestion
            dtype=ManimColorDType,
```
Or didn't we have something else similar?